### DISCUSSION TITLE
Jam Track Audio Streams Encrypted

### DISCUSSION THREAD

**Discussion by Alex1331xela** [LeleDerGrasshalmi/FortniteEndpointsDocumentation#343] [Q&A] [answered]
Hi, I'm noticing that for [Jam Tracks (Sparks tracks)](https://github.com/LeleDerGrasshalmi/FortniteEndpointsDocumentation/blob/main/EpicGames/FN-Content/Explanations/SparkTracks.md)'s audio, preview audios (pid) are not encrypted, but for the other types (sid, stereoId, instrumentalId) the audio streams seem to be encrypted with ClearKey behind a "key.key" (as seen in the track's [b64 encrpted xml](https://github.com/LeleDerGrasshalmi/FortniteEndpointsDocumentation/discussions/59#discussioncomment-7807930)). I was just wondering if you'd have any knowledge/documentation on this? Thanks so much!

**Comment by StupidRepo** [comment]
Oh I wish this had an answer, I want some answers for this too!

**Reply by Alex1331xela** [reply]
Unfortunately, I'm not too familiar with ClearKey encryption, or where one would find the key to decrypt these audio streams 😔

I'm assuming the game would use the decryption key to download a jam track's audio streams on the fly (for use in the jam emotes or in the Festival modes), however I'm not sure how one would go about finding out what that key is.

**Reply by StupidRepo** [reply]
Well, there are [AES keys archived](https://github.com/dippyshere/fortnite-aes-archive) for each version of Fortnite however I have no idea how to decrypt stuff with AES... And I have no idea what ClearKey is either... 😓

**Reply by LeleDerGrasshalmi** [reply, marked as answer]
hi @Alex1331xela & @StupidRepo,

these are not encrypted using the games aes key(s).

the algorithm for decrypting streamed audio/videos is not public and wont be public. These are encrypted for legal (copyright) stuff - not much you can do other than reverse engineer it yourself.
